<?xml version="1.0"?>
<!--
     Example launch for testing live cooperation of radar and laserscanner
-->

<launch>
  <arg name="record_data" default="false" />

  <node pkg="tf" type="static_transform_publisher" name="map_radar" args="0 0 0 0 0 0 1 map radar 100" />
  <node pkg="tf" type="static_transform_publisher" name="map_laser" args="0 0 0 0 0 0 1 map laser 100" />

  <!-- STARTING MRS61xx - modify your ip address -->

  <node name="sick_mrs_6xxx" pkg="sick_scan" type="sick_generic_caller" respawn="false" output="screen">
    <param name="filter_echos" type="int" value="0" />
    <param name="scanner_type" type="string" value="sick_mrs_6xxx"/>
    <param name="range_min" type="double" value="0.1" />
    <param name="range_max" type="double" value="250.0" />
    <param name="hostname" type="string" value="192.168.0.24" />
    <param name="port" type="string" value="2112" />
    <param name="timelimit" type="int" value="5" />
    <param name="min_ang" type="double" value="-1.047197333" />
    <param name="max_ang" type="double" value="+1.047197333" />

  </node>

  <node name="sick_rms_3xx" pkg="sick_scan" type="sick_generic_caller" respawn="false" output="screen">
      <param name="range_min" type="double" value="0.1" />
      <param name="range_max" type="double" value="250.0" />
      <param name="scanner_type" type="string" value="sick_rms_3xx"/>
      <param name="hostname" type="string" value="192.168.0.232" />
      <param name="port" type="string" value="2112" />

      <param name="transmit_raw_targets" type="bool" value="true" />
      <param name="transmit_objects" type="bool" value="true" />

      <param name="emul_sensor" type="bool" value="false" />

      <!-- tracking_mode 0: BASIC-Tracking - use for tracking smaller objects -->
      <!-- tracking_mode 1: TRAFFIC-Tracking - use for tracking larger objects like vehicles -->
      <param name="tracking_mode" type="int" value="0" />

  </node>

<?ignore

  <arg name="rviz"    default="true" />
  <arg name="rviz_file" default="$(find sick_scan)/launch/rviz/combi.rviz" />
  <node pkg="rviz" type="rviz" name="$(anon rviz)" args="-d $(arg rviz_file)" output="screen" if="$(arg rviz)"/>

 <node pkg="rosbag" type="record" name="rosbag_record_scanner_radar"
       args="record -o ~/rosbag_data/combi -a --lz4"
       if="$(arg record_data)" />

 <node pkg="rosbag" type="record" name="rosbag_record_scanner_radar"
       args="record -a --lz4"/>


  <node pkg="sick_scan" type="pcl_converter" name="pcl_converter" args="/cloud /image" output="screen" />
  <node pkg="sick_scan" type="radar_object_marker" name="radar_object_marker" />
?>
<?ignore
rosrun rosbag record record -o /mnt/hgfs/developnt/ros/bags/combi -a --lz4 -x /tf -->
rosbag play /mnt/hgfs/development/ros/bags/combi_2018-07-15-15-48-16.bag -l -->
?>

</launch>
